src/Planners/include/BipedalLocomotion/Planners/UnicycleTrajectoryPlanner.h

```diff
@@ -0,0 +1,234 @@
+/**
+ * @file UnicycleTrajectoryPlanner.h
+ * @authors Lorenzo Moretti, Diego Ferigo, Giulio Romualdi, Stefano Dafarra
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_UNICYCLE_PLANNER_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_UNICYCLE_PLANNER_H
+
+#include <BipedalLocomotion/Contacts/ContactPhaseList.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+
+#include <deque>
+#include <iDynTree/Model.h>
+#include <iDynTree/VectorDynSize.h>
+
+#include <memory>
+
+#include <UnicycleGenerator.h>
+#include <UnicyclePlanner.h>
+
+#include <CoMHeightTrajectoryGenerator.h>
+#include <vector>
+
+namespace BipedalLocomotion::Planners
+{
+class UnicycleTrajectoryPlanner;
+struct UnicycleTrajectoryPlannerInput;
+struct UnicycleTrajectoryPlannerOutput;
+struct UnicycleTrajectoryPlannerParameters;
+} // namespace BipedalLocomotion::Planners
+
+struct BipedalLocomotion::Planners::UnicycleTrajectoryPlannerInput
+{
+    /*
+    if UnicycleController::PERSON_FOLLOWING, the plannerInput is a vector of size 2 (i.e., [x, y])
+    if UnicycleController::DIRECT, the plannerInput is a vector of size 3 (i.e., [xdot, ydot, wz])
+    */
+    Eigen::VectorXd plannerInput; // The input to the unicycle planner.
+
+    DCMInitialState dcmInitialState; // The initial state of the DCM trajectory generator.
+
+    struct COMInitialState
+    {
+        Eigen::Vector2d initialPlanarPosition; // The initial planar position of the CoM.
+        Eigen::Vector2d initialPlanarVelocity; // The initial planar velocity of the CoM.
+    };
+
+    COMInitialState comInitialState; // The initial state of the CoM.
+
+    bool isLeftLastSwinging; // True if the last foot that was swinging is the left one. False
+                             // otherwise.
+
+    double initTime; // The initial time of the trajectory.
+
+    manif::SE3d measuredTransform; // The measured transform of the last foot that touched
+                                   // the floor.
+
+    static UnicycleTrajectoryPlannerInput generateDummyUnicycleTrajectoryPlannerInput();
+};
+
+struct BipedalLocomotion::Planners::UnicycleTrajectoryPlannerOutput
```

Comment:
look at the output I was expecting the function to return a ContactPhaseList instead of a deque of steps. Is this wanted?

Reply:
That's clear!

Reply:
with commit 212a07cfab3adae0f152f7b6de4c49dd9e2507d3 the `contactPhaseList` has been removed from the `UnicycleTrajectoryPlannerOutput`. The idea is that `UnicycleTrajectoryPlannerOutput` contains only information needed by the `UnicycleTrajectoryGenerator` of #845.

User can still the `contactPhaseList` returned by the `UnicycleTrajectoryPlannerOutput` leveraging the added `getContactPhaseList` member function.